src/main/java/Backend/ComparatorExpression.java

```diff
@@ -9,19 +9,47 @@
 public class ComparatorExpression extends Expression {
 
     // if x^2 + 5 >= 0 is to be stored, then expr would store the expression 'x^2 + 5'
-    private final Expression expr;
+    private final Expression lExpression;
+    private final Expression rExpression;
 
-    public ComparatorExpression(Expression expr){
-        super(">=");
-        this.expr = expr;
+    public ComparatorExpression(String comparatorOp, Expression lExpression, Expression rExpression){
+        super(comparatorOp);
+        this.lExpression = lExpression;
+        this.rExpression = rExpression;
     }
 
     @Override
     public double evaluate(Map<String, Double> arguments) {
-        double greaterThan;
-        if (expr.evaluate(arguments) >= 1) {greaterThan = 1;}
-        else {greaterThan = -1;}
+        boolean comparisonHolds;
+        double pixelValue;
 
-        return greaterThan;
+        double lExpressionVal = this.lExpression.evaluate(arguments);
+        double rExpressionVal = this.rExpression.evaluate(arguments);
+
+        switch(getItem()){
+            case ">=":
+                comparisonHolds = lExpressionVal >= rExpressionVal;
+                break;
+            case "<=":
+                comparisonHolds = lExpressionVal <= rExpressionVal;
+                break;
+            case ">":
+                comparisonHolds = lExpressionVal > rExpressionVal;
+                break;
+            case "<":
+                comparisonHolds = lExpressionVal < rExpressionVal;
+                break;
+            case "==":
+                comparisonHolds = lExpressionVal == rExpressionVal;
```

Comment:
Being doubles rather than checking for equality, we might want to check that they are within some threshold of each other (floating point errors and all that)

Reply:
Good point, you can slap a todo onto it, or I can later. I can work on that with my next batch of changes to ComparatorExpression.